frontend/src/components/QueryBuilder/QueryBuilder.tsx

```diff
@@ -1919,184 +1565,38 @@ function QueryBuilder<T extends SearchParamsInput | ErrorSearchParamsInput>(
 		],
 	)
 
-	const [paginationToUrlParams, setPaginationToUrlParams] = useQueryParams({
-		page: NumberParam,
-	})
-	useEffect(() => {
-		if (page !== undefined) {
-			setPaginationToUrlParams({ page }, 'replaceIn')
-		}
-	}, [setPaginationToUrlParams, page])
-
-	// Set initial state from URL params.
-	const [initialized, setInitialized] = useState(false)
-	useEffect(() => {
-		setPage(paginationToUrlParams.page || START_PAGE)
-
-		if (searchParamsToUrlParams.query !== undefined) {
-			setSearchParams(searchParamsToUrlParams as T)
-		} else {
-			setSearchParams(emptySearchParams)
-		}
-
-		setInitialized(true)
-		// Only want this to run on init.
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [])
-
-	const [forceReload, setForceReload] = useQueryParam('reload', BooleanParam)
-	useEffect(() => {
-		if (forceReload && searchParamsToUrlParams.query !== undefined) {
-			setSearchParams(searchParamsToUrlParams as T)
-			setForceReload(false)
-		}
-	}, [forceReload, searchParamsToUrlParams, setForceReload, setSearchParams])
-
-	useEffect(() => {
-		if (!segmentsLoading) {
-			if (activeSegmentUrlParam) {
-				selectSegment(activeSegmentUrlParam)
-			}
-			if (searchParamsToUrlParams.query !== undefined) {
-				setSearchParams(searchParamsToUrlParams as T)
-			}
-		}
-		// We only want to run this once after loading segments.
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [segmentsLoading])
-
-	// Errors Segment Deep Linking
-	useEffect(() => {
-		if (selectedSegment && selectedSegment.id && selectedSegment.name) {
-			if (!isEqual(activeSegmentUrlParam, selectedSegment)) {
-				setActiveSegmentUrlParam(selectedSegment, 'replaceIn')
-			}
-		} else if (activeSegmentUrlParam !== undefined) {
-			setActiveSegmentUrlParam(undefined, 'replace')
-		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [selectedSegment, setActiveSegmentUrlParam])
-
-	// Main useEffect that handles updates to URL on every filter change.
-	useEffect(() => {
-		if (initialized && searchParams.query !== undefined) {
-			setSearchParamsToUrlParams(
-				normalizeParams(searchParams),
-				'replaceIn',
-			)
-		}
-	}, [
-		setSearchParamsToUrlParams,
-		searchParams,
-		selectedSegment,
-		activeSegmentUrlParam,
-		initialized,
-	])
-
-	useEffect(() => {
```

Comment:
the sync button wasn't working previously because of some changes to the relative date logic